### ISSUE TITLE
There's no blobl option currently for batch-aware structured metadata queries

### ISSUE THREAD

**Issue by mihaitodor** [benthosdev/benthos#1882]
Here's a sample config:

```yaml
input:
  generate:
    interval: 0s
    mapping: meta test = count("foo")
    count: 2
    batch_size: 2
  processors:
    - log:
        message: ${! @.test.from(-1) } - ${! meta("test").from(-1) }

output:
  drop: {}
```

It produces the following logs:
```
INFO 1 - 2                                         @service=benthos label="" path=root.input.processors.0
INFO 2 - 2                                         @service=benthos label="" path=root.input.processors.0
```

Also, if this gets addressed, I guess [the docs](https://www.benthos.dev/docs/guides/bloblang/methods#from) for this method should be updated to mention structured metadata too

**Comment by Jeffail** [comment]
My bad, I was wrong in the discord chan. The new `@` syntax is the structured equivalent of `root_meta` rather than `meta` as it reflects changes being made within the mapping itself and therefore only refers to the metadata being formed for the current message.

In order to support batch processing of structured metadata values I'll need to add a new metadata function, and we ought to deprecate `meta` and `root_meta`. I think I avoided it originally as I figured it was a niche use case and I couldn't come up with a better name than just `meta`.
